app/src/main/java/com/sunasterisk/travelapp/data/repository/RestaurantRepositoryImpl.kt

```diff
@@ -0,0 +1,95 @@
+package com.sunasterisk.travelapp.data.repository
+
+import com.sunasterisk.travelapp.data.OnDataCallback
+import com.sunasterisk.travelapp.data.models.Restaurant
+import com.sunasterisk.travelapp.data.source.RestaurantDatasource
+import com.sunasterisk.travelapp.utils.ApiEndpoint
+import com.sunasterisk.travelapp.utils.ApiEndpoint.LOCATION_ID
+import com.sunasterisk.travelapp.utils.ApiEndpoint.MEAL_TYPE
+import com.sunasterisk.travelapp.utils.ApiEndpoint.RESTAURANT_TYPE
+import com.sunasterisk.travelapp.utils.KeyResponse.DATA
+import org.json.JSONException
+import org.json.JSONObject
+
+class RestaurantRepositoryImpl private constructor(
+    private val local: RestaurantDatasource.Local,
+    private val remote: RestaurantDatasource.Remote
+) : RestaurantRepository {
+    override fun searchRestaurantsByProperty(
+        latitude: String,
+        longtitude: String,
+        mealType: String?,
+        restaurantType: String?,
```

Comment:
gán giá trị mặc định 
```suggestion
        mealType: String? = null,
        restaurantType: String? = null,
```